src/java.base/share/classes/java/util/zip/Deflater.java

```diff
@@ -49,29 +49,33 @@
  * thrown.
  * <p>
  * This class deflates sequences of bytes into ZLIB compressed data format.
- * The input byte sequence is provided in either byte array or byte buffer,
+ * The input byte sequence is provided in either byte array or {@link ByteBuffer},
  * via one of the {@code setInput()} methods. The output byte sequence is
- * written to the output byte array or byte buffer passed to the
+ * written to the output byte array or {@code ByteBuffer} passed to the
  * {@code deflate()} methods.
  * <p>
- * The following code fragment demonstrates a trivial compression
- * and decompression of a string using {@code Deflater} and
- * {@code Inflater}.
- * {@snippet id="compdecomp" lang="java" class="Snippets" region="DeflaterInflaterExample"}
+ * To release resources used by the {@code Deflater}, applications must close the
+ * {@code Deflater} by calling either the {@link #end()} or the {@link #close()} method.
+ * After the {@code Deflater} has been closed, subsequent calls to several methods
+ * of the {@code Deflater} will throw an {@link IllegalStateException}.
  *
  * @apiNote
- * To release resources used by this {@code Deflater}, the {@link #end()} method
- * should be called explicitly. Subclasses are responsible for the cleanup of resources
- * acquired by the subclass. Subclasses that override {@link #finalize()} in order
- * to perform cleanup should be modified to use alternative cleanup mechanisms such
- * as {@link java.lang.ref.Cleaner} and remove the overriding {@code finalize} method.
+ * This class implements {@link AutoCloseable} to facilitate its usage with
+ * {@code try}-with-resources statement. The {@linkplain Deflater#close() close() method} simply
+ * calls {@code end()}. Subclasses should override {@linkplain #end()} to clean up the
+ * resources acquired by the subclass.
```

Comment:
I wonder if the note for subclasses should move to the end method as implSpec, it looks out of place in the class API note.

Reply:
I believe an `@implSpec` only restricts the implementation of this immediate method, such as those seen on default methods, and has no impact on the overrides. If we need to specify something for overrides to abide to, I believe they usually stay in the API specification itself.

Reply:
> I wonder if the note for subclasses should move to the end method as implSpec, it looks out of place in the class API note.

Stuart, in his inputs here https://github.com/openjdk/jdk/pull/19675#issuecomment-2172433871 and here https://github.com/openjdk/jdk/pull/19675#discussion_r1848945718 had noted that it would be useful to add this guidance to subclasses. But I see that he wasn't fully convinced whether it must be a apiNote or something else. Given what you note about moving this detail for subclasses to a `@implSpec` on the end() method, I have now updated the PR accordingly.

Reply:
Hello Chen,
> I believe an @implSpec only restricts the implementation of this immediate method, such as those seen on default methods, and has no impact on the overrides.

Based on the details about these tags here https://openjdk.org/jeps/8068562, I think the `@implSpec` is the most relevant one for this text:

> Interface implementors or class subclassers use the information here in order to decide whether it is sensible or necessary to override a particular method